File: api/src/main/java/com/api/app/entity/PointHistory.java
package com.api.app.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;
import org.apache.ibatis.type.Alias;

import java.time.LocalDateTime;

/**
 * @author system
 * @version 1.0
 * @since 2025-10-27
 */
@Alias("PointHistory")
@Getter
@Setter
public class PointHistory extends SystemEntity {
    private static final long serialVersionUID = 1123456789012345678L;

    @Schema(description = "포인트기록번호")
    private String pointHistoryNo;

    @Schema(description = "회원번호")
    private String memberNo;

    @Schema(description = "금액")
    private Long amount;

    @Schema(description = "포인트적립사용코드")
    private String pointTransactionCode;

    @Schema(description = "포인트적립사용사유코드")
    private String pointTransactionReasonCode;

    @Schema(description = "포인트적립사용번호")
    private String pointTransactionReasonNo;

    @Schema(description = "시작일시")
    private LocalDateTime startDateTime;

    @Schema(description = "종료일시")
    private LocalDateTime endDateTime;
}

File: api/src/main/kotlin/com/api/app/entity/PointHistory.kt
package com.api.app.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.LocalDateTime

@Entity
@Table(name = "POINT_HISTORY")
class PointHistory : SystemEntity() {

    @Id
    @Column(name = "POINT_HISTORY_NO", length = 15, nullable = false)
    var pointHistoryNo: String = ""

    @Column(name = "MEMBER_NO", nullable = false, length = 15)
    var memberNo: String = ""

    @Column(name = "AMOUNT", nullable = false, columnDefinition = "numeric")
    var amount: Long = 0

    @Column(name = "POINT_TRANSACTION_CODE", nullable = false, length = 3)
    var pointTransactionCode: String = ""

    @Column(name = "POINT_TRANSACTION_REASON_CODE", nullable = false, length = 3)
    var pointTransactionReasonCode: String = ""

    @Column(name = "POINT_TRANSACTION_REASON_NO", length = 15)
    var pointTransactionReasonNo: String? = null

    @Column(name = "START_DATE_TIME")
    var startDateTime: LocalDateTime? = null

    @Column(name = "END_DATE_TIME")
    var endDateTime: LocalDateTime? = null

    @Column(name = "UPPER_POINT_HISTORY_NO", length = 15)
    var upperPointHistoryNo: String? = null

    @Column(name = "REMAIN_POINT", columnDefinition = "numeric")
    var remainPoint: Long = 0
}
